ceph_medic/util/configuration.py

```diff
@@ -335,6 +335,18 @@ def _parse(self, lines):
                 else:
                     self.hosts[host_group].append(host_item)
         self.expand_nodes()
+        self.filter_groups()
+
+    def filter_groups(self):
+        """
+        Removes any groups from self.nodes that do not exist in
+        ceph_medic.metadata. We want to do this because we don't care
+        about nodes or groups in an inventory that are not part of
+        the ceph cluster.
+        """
+        for group in self.nodes.keys():
+            if group not in metadata:
+                del self.nodes[group]
```

Comment:
you can't alter a dictionary on a loop :(

![2hc79xg](https://user-images.githubusercontent.com/317847/27658352-bcf4c71a-5c1d-11e7-8a33-830b6acae6b4.gif)